mavros/src/lib/enum_to_string.cpp

```diff
@@ -176,6 +176,20 @@ std::string to_string(MAV_TYPE e)
 
 	return mav_type_strings[idx];
 }
+
+MAV_TYPE mav_type_from_str(const std::string &mav_type)
+{
+	for (size_t idx = 0; idx < mav_type_strings.size(); idx++) {
+		if (mav_type_strings[idx] == mav_type) {
+			std::underlying_type<MAV_TYPE>::type rv = idx;
+			return static_cast<MAV_TYPE>(rv);
+		}
+	}
+
+	ROS_ERROR_STREAM_NAMED("uas", "TYPE: Unknown MAV_TYPE: " << mav_type);
+	return MAV_TYPE::GENERIC;
+}
+
 // [[[end]]] (checksum: 31488f5970b0f82b3efef71e32590bb6)
```

Comment:
Do not change generated code.

Also i think it is better to add enum entry name to value function.
For that another array is needed, or changes in current (use std::tuple).